pyvista/plotting/widgets.py

```diff
@@ -651,6 +715,132 @@ def callback(normal, loc):
+
+        normal : str or tuple(float), optional
+            The starting normal vector of the plane.
+
+        invert : bool, optional
+            Flag on whether to flip/invert the clip.
+
+        widget_color : ColorLike, optional
+            Either a string, RGB list, or hex color string.
+
+        value : float, optional
+            Set the clipping value along the normal direction.
+            The default value is 0.0.
+
+        assign_to_axis : str or int, optional
+            Assign the normal of the plane to be parallel with a given
+            axis.  Options are ``(0, 'x')``, ``(1, 'y')``, or ``(2,
+            'z')``.
+
+        tubing : bool, optional
+            When using an implicit plane wiget, this controls whether
+            or not tubing is shown around the plane's boundaries.
+
+        origin_translation : bool, optional
+            If ``False``, the plane widget cannot be translated by its
+            origin and is strictly placed at the given origin. Only
+            valid when using an implicit plane.
+
+        outline_translation : bool, optional
+            If ``False``, the box widget cannot be translated and is
+            strictly placed at the given bounds.
+
+        implicit : bool, optional
+            When ``True``, a ``vtkImplicitPlaneWidget`` is used and
+            when ``False``, a ``vtkPlaneWidget`` is used.
+
+        normal_rotation : bool, optional
+            Set the opacity of the normal vector arrow to 0 such that
+            it is effectively disabled. This prevents the user from
+            rotating the normal. This is forced to ``False`` when
+            ``assign_to_axis`` is set.
+
+        interaction_event : vtk.vtkCommand.EventIds, optional
+            The VTK interaction event to use for triggering the callback.
+
+        origin : tuple(float), optional
+            The starting coordinate of the center of the plane.
+
+        **kwargs : dict, optional
+            All additional keyword arguments are passed to
+            :func:`BasePlotter.add_volume` to control how the volume is
+            displayed. Only applicable if ``volume`` is either a
+            :class:`pyvista.UniformGrid` and :class:`pyvista.RectangularGrid`
+
+        Returns
+        -------
+        vtk.vtkPlaneWidget or vtk.vtkImplicitPlaneWidget
+            The VTK plane widget depending on the value of ``implicit``.
+
+        """
+        if isinstance(volume, (pyvista.UniformGrid, pyvista.RectilinearGrid)):
+            volume = self.add_volume(volume, **kwargs)
+        elif not isinstance(volume, pyvista.plotting.Volume):
+            raise TypeError(
+                'The `volume` parameter type must be either pyvista.UniformGrid, '
+                'pyvista.RectangularGrid, or a pyvista.plotting.volume.Volume '
+                'from `Plotter.add_volume`.'
+            )
+        else:
+            assert_empty_kwargs(**kwargs)
+
+        plane = _vtk.vtkPlane()
+        _widget = []
```

Comment:
This local reference to track the plane widget seems weird to me... 

What happens if we add multiple volumes clipped by planes or add multiple planes clipping one volume?

Reply:
Could we use `GetClippingPlanes` somehow from the mapper to track this instead?

Reply:
> This local reference to track the plane widget seems weird to me...
What happens if we add multiple volumes clipped by planes or add multiple planes clipping one volume?

Multi-plane clipping works out, but I agree that it seems a bit weird here. Let me see if there's a better way to do this.

Reply:
Fixed in 5483bdb9dd6fce17603153e65a63c4a1ec9dd486

Multiple planes work:

https://user-images.githubusercontent.com/11981631/213025985-20b98e57-9fa1-4b52-821c-52d56a0ceb35.mp4